[doc] Add note about PWAs and Metro/expo-router (#24026)
expo/expo: docs/pages/guides/progressive-web-apps.mdx

A progressive web app (or PWA for short) is a website that can be installed on the user's device and used offline. If you build your native app with Expo, then Expo CLI can generate a lot of the PWA automatically based on how the native app works. Ex: icons, splash screens, orientation, and so on. [Add service workers](https://expo.fyi/enabling-web-service-workers) to get a complete PWA.

> > This guide is oriented towards apps using webpack. If you use [Metro for web](/guides/customizing-metro.mdx#web-support) (such as with [Expo Router](/routing/introduction.mdx)), you can still build a PWA, but the Expo tooling suggested in this guide is not all intended for it. You likely want to use the default `web.output` configuration of `single` to generate a SPA, and then you can modify your **public/index.html** to add meta tags and PWA assets, and use [expo-router/head](/router/reference/static-rendering.mdx#meta-tags) to set meta tags on each route. [Workbox](https://web.dev/learn/pwa/workbox/) is recommended for service workers.

Reviewer: ```suggestion
> This guide is oriented towards apps using webpack. If you use [Metro for web](/guides/customizing-metro/#web-support) (such as with [Expo Router](/routing/introduction)), you can still build a PWA. However, the Expo tooling suggested in this guide is not all intended for it. Use the default [`web.output`](/versions/latest/config/app/#output) property with `single` as its value to generate a SPA. You can then modify your **public/index.html** to add meta tags and PWA assets, and use [`expo-router/head`](/router/reference/static-rendering/#meta-tags) to set meta tags on each route. [Workbox](https://web.dev/learn/pwa/workbox/) is recommended for service workers.
```

- There is no need to use file extension such as `.mdx` for internal links.
- "...in this guide is not all intended for it":  it is unclear which parts are intended and which aren't 
- "...likely want to use...": Can we say "You'll have to use the default..."? If we are providing a solution here, we'll have to be sure about it.
